Libplanet/Net/Protocols/RoutingTable.cs

```diff
@@ -250,37 +299,80 @@ internal void AddPeer(BoundPeer peer, DateTimeOffset updated)
                 );
             }
 
-            _logger.Debug(
-                "Adding a peer {Peer} to peer {Address}'s routing table...",
-                peer,
-                _address);
-            BucketOf(peer).AddPeer(peer, updated);
+            if (_staticPeerStates.ContainsKey(peer.Address))
+            {
+                _logger.Verbose("The peer {Peer} to be added is already in static peer.", peer);
+
+                // This done because peer's other attribute except public key might be changed.
+                // (eg. public IP address, endpoint)
+                PeerState state = _staticPeerStates[peer.Address];
+                state.Peer = peer;
+                state.LastUpdated = updated;
+            }
+            else
+            {
+                _logger.Debug(
+                    "Adding a peer {Peer} to peer {Address}'s routing table...",
+                    peer,
+                    _address);
+                BucketOf(peer).AddPeer(peer, updated);
+            }
         }
 
+        // TODO
```

Comment:
Could you describe what should do in this section as additional comment?

Reply:
Was mistake, removed! Thanks for the review.